transport/http/handler.go

```diff
@@ -59,6 +59,11 @@ func (h handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
+	if err == context.DeadlineExceeded {
+		http.Error(w, "remote timeout", http.StatusGatewayTimeout)
```

Comment:
After many discussion in person, we decided to have different types of timeout behind the TimeoutError interface:

 - handlerTimeoutError for representing a timeout error on the server side. It includes caller, service, procedure and the time waited.
 - remoteTimeoutError for the client representation of a handlerTimeoutError.
 - clientTimeoutError, for representing a timeout error on the client side, it includes the service, procedure and time waited.

Because TChannel returns to the client the same error code in both client side and server side timeouts, all timeout errors via TChannel will be represented as a remoteTimeoutError. Overtime, we can work to improve the situation in TChannel (at least more informative error messages).